src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1474,37 +1476,53 @@ void Graph::Infer(SyncInferRequest* request) {
 void Graph::SortTopologically() {
     OV_ITT_SCOPE(FIRST_INFERENCE, itt::domains::intel_cpu_LT, "Graph::SortTopologically");
 
+    for (auto &node : graphNodes) {
+        // Sort in / out Parent edges by port index
+        // Make first N (N == port_num) edge indexes match with port index
+       std::sort(node->parentEdges.begin(), node->parentEdges.end(), [](const EdgeWeakPtr& lhs, const EdgeWeakPtr& rhs) {
+           auto lhs_ptr = lhs.lock();
+           auto rhs_ptr = rhs.lock();
+           OPENVINO_ASSERT(lhs_ptr && rhs_ptr);
+           return lhs_ptr->getOutputNum() < rhs_ptr->getOutputNum();
+        });
+
+        // Set execIndex of all nodes to default invaild value
+        node->execIndex = -1;
+    }
+
     auto sort = [](const std::vector<NodePtr>& nodes) {
-        std::unordered_set<NodePtr> visited;
-        visited.reserve(nodes.size());
         std::vector<NodePtr> sorted;
         sorted.reserve(nodes.size());
 
+        int execIndexCnt = -1;
+
         std::function<void(const NodePtr)> visit;
-        visit = [&visited, &sorted, &visit](const NodePtr node) {
-            const bool inserted = visited.insert(node).second;
-            if (!inserted)
+        visit = [&execIndexCnt, &sorted, &visit](const NodePtr node) {
+            if (node->execIndex >= 0)
```

Comment:
Could you please clarify how it is supposed to work?
Maybe I didn't get something
I mean how the `execIndexCnt` replaces the `std::unordered_set<NodePtr> visited`.
Imagine the following sequence:
1. SortTopologically called
2. SomeGraphTransformation called and changed the graph
3. SortTopologically called

So, here we are on stage 3.
The `node->execIndex` is already set for all the nodes in stage 1.
The stage 2 could have changed the graph order.
But after your changes, we won't process any nodes to update the order, because of this check.

Reply:
The execIndex of all the nodes are set to -1 to indicate it is not visited in this turn of sorting.

Reply:
My bad, missed this part